h2o-core/src/main/java/hex/grid/GridSearch.java

```diff
@@ -245,6 +245,17 @@ private void gridSearch(Grid<MP> grid) {
           _job.update(1);
           // Always update grid in DKV after model building attempt
           grid.update(_job);
+          final MP generalParams = _hyperSpaceWalker.getParams();
+          if (generalParams._export_checkpoints_dir != null) {
+            final String gridPath = generalParams._export_checkpoints_dir + "/" + grid._key;
+            try {
+              grid.exportBinary(gridPath);
+              // Models are exported automatically when export_checkpoints_dir is defined
+            } catch (IOException e) {
```

Comment:
is this try-catch necessary here? since we are basically ignoring the exception what is the point of throwing it?

Reply:
## My POV
From the point of `exportBinary` method, it gives the caller more freedom to decide what to do when the export fails. But with power comes great responsibility to catch the exceptions and treat them "properly".

In this case, we only log something went bad, because taking down the whole grid only because it could not be exported is not appropriate. Maybe where will be other handlings in future. 

## Conclusion
As there is no other use case for it, I adopted your logic and removed the try catch. The `exportBinary` method now logs this itself.

Reply:
From the point of exportBinary method, it gives the caller more freedom to decide what to do when the export fails. But with power comes great responsibility to catch the exceptions and treat them "properly".

In this case, we only log something went bad, because taking down the whole grid only because it could not be exported is not appropriate. Maybe where will be other handlings in future.

It's also used in `exportGrid` in `GridImportExportHandler` - which doesn't catch the exception, but rather propagates it all the way to the client.